jetson/ra_kinematics/mrover_arm.cpp

```diff
@@ -0,0 +1,747 @@
+            else if (sim_mode) {
+                arm_state.set_joint_angles(target_angles);
+            }
+
+            if (control_state != ControlState::EXECUTING) {
+                send_kill_cmd();
+            }
+
+            std::this_thread::sleep_for(std::chrono::milliseconds((int) SPLINE_WAIT_TIME));
+        }
+        else {
+            std::this_thread::sleep_for(std::chrono::milliseconds(200));
+            spline_t = 0;
+        }   
+    }
+}
+
+void MRoverArm::simulation_mode_callback(std::string channel, SimulationMode msg) {
+    sim_mode = msg.sim_mode;
+    std::cout << "Received Simulation Mode value: " << sim_mode << "\n";
+}
+
+void StandardArm::lock_joints_callback(std::string channel, LockJoints msg) {
+    std::cout << "Running lock_joints_callback:   ";
+
+    arm_state.set_joint_locked(0, (bool) msg.jointa);
+    arm_state.set_joint_locked(1, (bool) msg.jointb);
+    arm_state.set_joint_locked(2, (bool) msg.jointc);
+    arm_state.set_joint_locked(3, (bool) msg.jointd);
+    arm_state.set_joint_locked(4, (bool) msg.jointe);
+    arm_state.set_joint_locked(5, (bool) msg.jointf);
+
+    std::cout << "\n";
+}
+
+void MRoverArm::zero_position_callback(std::string channel, ZeroPosition msg) {
+    for (size_t i = 0; i < arm_state.num_joints(); ++i) {
+        if (i == 1) {
+            arm_state.set_joint_encoder_offset(i, arm_state.get_joint_angle(i) + 1.09);
+        }
+        else {
+            arm_state.set_joint_encoder_offset(i, arm_state.get_joint_angle(i));
+        }
+    }
+}
+
+void MRoverArm::encoder_angles_sender() {
+    // Continuously send mock values if in sim mode
+    while (true) {
+        if (sim_mode) {
+            encoder_angles_sender_mtx.lock();
+
+            ArmPosition arm_position;
+            arm_position.joint_a = arm_state.get_joint_angle(0);
+            arm_position.joint_b = arm_state.get_joint_angle(1);
+            arm_position.joint_c = arm_state.get_joint_angle(2);
+            arm_position.joint_d = arm_state.get_joint_angle(3);
+            arm_position.joint_e = arm_state.get_joint_angle(4);
+            arm_position.joint_f = arm_state.get_joint_angle(5);
+            lcm_.publish("/arm_position", &arm_position);
+
+            encoder_angles_sender_mtx.unlock();
+        }
+
+        std::this_thread::sleep_for(std::chrono::milliseconds(SPLINE_WAIT_TIME));
+    }
+}
+
+void MRoverArm::publish_config(const std::vector<double> &config, std::string channel) {
+    ArmPosition arm_position;
+
+    if (arm_state.num_joints() == 6) {
+        arm_position.joint_a = config[0];
+        arm_position.joint_b = config[1];
+        arm_position.joint_c = config[2];
+        arm_position.joint_d = config[3];
+        arm_position.joint_e = config[4];
+        arm_position.joint_f = config[5];
+    }
+    else {
+        arm_position.joint_a = config[0];
+        arm_position.joint_b = config[1];
+        arm_position.joint_c = config[2];
```

Comment:
When does it enter this else statement? Is it for the science arm? If so, then the members should be double joint_a, b, c, and e. (e not d)